broker/lib/middleware.js

```diff
@@ -44,6 +46,38 @@ exports.validateCreateRequest = function () {
   };
 };
 
+exports.validateSchemaForRequest = function (target, operation) {
+  return function (req, res, next) {
+    const plan = getPlanFromRequest(req);
+
+    const schema = _.get(plan, `schemas.${target}.${operation}.parameters`);
+
+    if (schema) {
+      const parameters = _.get(req, 'body.parameters', {});
+
+      const schemaVersion = schema['$schema'] || '';
+      const validator = new ajv({ schemaId: 'auto' });
+      if (schemaVersion.includes('draft-06')) {
+        validator.addMetaSchema(require('ajv/lib/refs/json-schema-draft-06.json'));
+      } else if (schemaVersion.includes('draft-04')) {
+        validator.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));
```

Comment:
From [ajv documentation](https://github.com/epoberezkin/ajv#using-version-6),

```
To use Ajv with draft-04 schemas in addition to explicitly adding meta-schema you also need to use option schemaId:
var ajv = new Ajv({schemaId: 'id'});
```
Has this been considered here?

Reply:
Yes, this is handled by `new ajv({ schemaId: 'auto' });` (I just see that this should be PascalCase). This will use `id` for v4 and `#id` for v6 and v7.

Reply:
Ohh, I had missed that statement! (y)